katsdpcontroller/scheduler.py

```diff
@@ -2640,6 +2647,18 @@ def statusUpdate(self, driver, status):
                     # The task itself is responsible for advancing to
                     # READY
             task.set_status(status)
+            if (task.kill_sent_time is not None
+                    and status.state != 'TASK_KILLING'
+                    and status.state not in TERMINAL_STATUSES
+                    and self._loop.time() > task.kill_sent_time + self.kill_timeout):
+                logger.warning('Retrying kill on %s (task ID %s)',
+                               task.name, status.task_id.value)
+                self._driver.killTask(status.task_id)
+        else:
+            if status.state not in TERMINAL_STATUSES:
+                logger.warning('Received status update for unknown task %s, killing it',
+                               status.task_id.value)
```

Comment:
Is the timestamp field in the status of any use ? i.e. does it tell us when the task was launched or only when this update was sent. Just wondering if there is any further debug info we can put into the warning to try and track down why we have an unknown task.

Reply:
I think the timestamp will just tell us when the update was sent (or maybe when the reconciliation request was received). There probably isn't much else we can get:

> Status updates that result from reconciliation requests will their reason field set to REASON_RECONCILIATION. Note that most of the other fields in the returned TaskStatus message will not be set: for example, reconciliation cannot be used to retrieve the labels or data fields associated with a running task.